### PR TITLE
decompiled unk_02097720.s

### PR DESCRIPTION
**Pull request by dyedquartz**

### CODE REVIEW

File: src/unk_02097720.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#include "unk_02097720.h"
+#include "unk_0203E348.h"
+
+void sub_02097720(TaskManager *taskManager, u16* unknownPtr) {
+    FieldSystem *fsys = TaskManager_GetSys(taskManager);
+    Unknown_02097720 *data = AllocFromHeapAtEnd(0xb, sizeof(Unknown_02097720));
+    MI_CpuFill8(data, 0, 0xc);
```
Comment (PikalaxALT) [inline src/unk_02097720.c:7]:
```suggestion
    MI_CpuClear8(data, sizeof(Unknown_02097720));
```

File: include/unk_02097720.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#ifndef HEARTGOLD_UNK_02097720_H
+#define HEARTGOLD_UNK_02097720_H
+
+#include "task.h"
+
+typedef struct Unknown_02097720 {
+    u32 unknown_0;
+    u16* unknown_1;
```
Comment (PikalaxALT) [inline include/unk_02097720.h:8]: Field names, if unknown, should reflect the byte offset within the struct (in this case, 4).

File: src/unk_02097720.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+BOOL sub_02097754(TaskManager *taskManager) {
+    FieldSystem *fsys = TaskManager_GetSys(taskManager);
+    Unknown_02097720* env = TaskManager_GetEnv(taskManager);
+
+    switch (env->unknown_0) {
+    case 0:
+        env->unknown_2 = sub_0203EB64(fsys);
+        env->unknown_0 += 1;
```
Comment (PikalaxALT) [inline src/unk_02097720.c:20]:
```suggestion
        env->unknown_0++;
```

File: src/unk_02097720.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        break;
+    case 1:
+        if(FieldSys_ApplicationIsRunning(fsys) == FALSE) {
```
Comment (PikalaxALT) [inline src/unk_02097720.c:23]: space after `if`, `while`, etc.

File: src/unk_02097720.c [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            (*env->unknown_1) = env->unknown_2[0xe];
```
Comment (PikalaxALT) [inline src/unk_02097720.c:24]:
```suggestion
            *env->unknown_1 = env->unknown_2[0xe]; // env->unknown_2 could be a struct
```

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: src/unk_02097720.c [hunk 6]
```diff
@@ -0,0 +1,40 @@
+#include "unk_02097720.h"
+#include "unk_0203E348.h"
+
+void sub_02097720(TaskManager *taskManager, u16* unknownPtr) {
+    FieldSystem *fsys = TaskManager_GetSys(taskManager);
+    Unknown_02097720 *data = AllocFromHeapAtEnd(0xb, sizeof(Unknown_02097720));
```
Comment (PikalaxALT) [inline src/unk_02097720.c:6]:
```suggestion
    Unknown_02097720 *data = AllocFromHeapAtEnd(11, sizeof(Unknown_02097720));
```

If it's a small number, represent it as decimal.
If there's a defined enum or constant for this value which fits the context, use that instead.
If it's a flag array and the bitmasks are named, use those bitmasks separated by the bitwise or operator (`|`).

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

**Review by PikalaxALT — DISMISSED** [review]

File: include/unk_02097720.h [hunk 7]
```diff
@@ -0,0 +1,15 @@
+#ifndef HEARTGOLD_UNK_02097720_H
+#define HEARTGOLD_UNK_02097720_H
+
+#include "task.h"
+#include "overlay_02.h"
+
+typedef struct UnkStruct_02097720 {
+    u32 unknown_0;
+    u16* unknown_4;
+    UnkStruct_Ov02_0224E4EC* unknown_8; // most likely a struct pointer
```
Comment (tgsm) [inline include/unk_02097720.h:10]: This comment is redundant

File: include/overlay_02.h [hunk 8]
```diff
@@ -7,6 +7,34 @@
...
+    u8 unk17;
+    u8 unk18;
+    u8 unk19;
+    u8 unk1A;
+    u8 unk1B;
+    u16 unk1C; // known from what's decompiled
+    u8 *unk20; // points to fsys->unk_111
+} UnkStruct_Ov02_0224E4EC;
+
+
```
Comment (tgsm) [inline include/overlay_02.h:37]: Extra newline

**Review by tgsm — CHANGES_REQUESTED** [review]

**Review by tgsm — DISMISSED** [review]

**Pull request merged**